tests/tuple-tree-generator/typescript/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+file(MAKE_DIRECTORY "${CMAKE_BINARY_DIR}/ts-ttg-test")
+
+set(TYPESCRIPT_TEST_DIR
+    "${CMAKE_SOURCE_DIR}/tests/tuple-tree-generator/typescript")
+configure_file("${TYPESCRIPT_TEST_DIR}/test.ts"
+               "${CMAKE_BINARY_DIR}/ts-ttg-test/test.ts" COPYONLY)
+configure_file(
+  "${TYPESCRIPT_TEST_DIR}/run-ts-roundtrip-test.sh"
+  "${CMAKE_BINARY_DIR}/ts-ttg-test/run-ts-roundtrip-test.sh" COPYONLY)
+configure_file(
+  "${TYPESCRIPT_TEST_DIR}/prepare-ts-roundtrip-test.sh"
+  "${CMAKE_BINARY_DIR}/ts-ttg-test/prepare-ts-roundtrip-test.sh" COPYONLY)
+configure_file("${TYPESCRIPT_TEST_DIR}/package.json"
+               "${CMAKE_BINARY_DIR}/ts-ttg-test/package.json" COPYONLY)
+configure_file("${TYPESCRIPT_TEST_DIR}/tsconfig.json"
+               "${CMAKE_BINARY_DIR}/ts-ttg-test/tsconfig.json" COPYONLY)
+
+add_custom_command(
+  OUTPUT "${CMAKE_BINARY_DIR}/ts-ttg-test/out/model.js"
+  COMMAND "${CMAKE_BINARY_DIR}/ts-ttg-test/prepare-ts-roundtrip-test.sh"
+          "${CMAKE_BINARY_DIR}/lib/typescript/model.ts"
+  DEPENDS generate-revngModel-tuple-tree-code)
+
+add_custom_target(generate-model_js ALL
+                  DEPENDS "${CMAKE_BINARY_DIR}/ts-ttg-test/out/model.js")
+
+macro(artifact_handler CATEGORY INPUT_FILE CONFIGURATION OUTPUT TARGET_NAME)
```

Comment:
Defining `artifact_handler` without a call to `register_derived_artifacts` doesn't make sense.
What were you trying to do?

Reply:
I want to run round-trip tests on all bitcode files, the script does:
1. run `revng model opt` on the bitcode to generate the model
2. run the typescript wrapper to deserialize the yaml and reserialize it
3. check with `revng model diff` that nothing changed, so we know that (2) worked correctly